Examples/Swift/ManageOfflineRegionsExample.swift

```diff
@@ -0,0 +1,190 @@
+        
+        if let packs = MGLOfflineStorage.shared.packs {
+            let pack = packs[indexPath.row]
+            
+            cell.textLabel?.text = "Region \(indexPath.row + 1): size: \(pack.progress.countOfBytesCompleted)"
+            cell.detailTextLabel?.text = "Percent completion: \(pack.progress.percentCompleted)%"
+
+        }
+
+        return cell
+
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let packs = MGLOfflineStorage.shared.packs else { return }
+        if let selectedRegion = packs[indexPath.row].region as? MGLTilePyramidOfflineRegion {
+            mapView.setVisibleCoordinateBounds(selectedRegion.bounds, animated: true)
+        }
+    }
+
+    @objc func startOfflinePackDownload() {
+
+        // Create a region that includes the current viewport and any tiles needed to view it when zoomed further in.
+        // Because tile count grows exponentially with the maximum zoom level, you should be conservative with your `toZoomLevel` setting.
+        let region = MGLTilePyramidOfflineRegion(styleURL: mapView.styleURL, bounds: mapView.visibleCoordinateBounds, fromZoomLevel: mapView.zoomLevel, toZoomLevel: mapView.zoomLevel + 2)
+
+        // Store some data for identification purposes alongside the downloaded resources.
+
+        let userInfo = ["name": "\(region.bounds)"]
+        let context = NSKeyedArchiver.archivedData(withRootObject: userInfo)
+
+        // Create and register an offline pack with the shared offline storage object.
+
+        MGLOfflineStorage.shared.addPack(for: region, withContext: context) { (pack, error) in
+            guard error == nil else {
+                // The pack couldn’t be created for some reason.
+                print("Error: \(error?.localizedDescription ?? "unknown error")")
+                return
+            }
+
+            // Start downloading.
+
+            pack!.resume()
+
+        }
+    }
+
+    // MARK: - MGLOfflinePack notification handlers
+
+    @objc func offlinePackProgressDidChange(notification: NSNotification) {
+
+        // Get the offline pack this notification is regarding,
+        // and the associated user info for the pack; in this case, `name = My Offline Pack`
+        if let pack = notification.object as? MGLOfflinePack,
+           let userInfo = NSKeyedUnarchiver.unarchiveObject(with: pack.context) as? [String: String] {
+
+            if pack.progress.countOfResourcesCompleted == pack.progress.countOfResourcesExpected {
+
+                let byteCount = ByteCountFormatter.string(fromByteCount: Int64(pack.progress.countOfBytesCompleted), countStyle: ByteCountFormatter.CountStyle.memory)
+
+                print("Offline pack “\(userInfo["name"] ?? "unknown")” completed: \(byteCount), \(pack.progress.countOfResourcesCompleted) resources")
+
+            }
+        }
+        self.tableView.reloadData()
+
+    }
+
+}
+
```

Comment:
Let's move all the methods that deal with setting up the `UITableView` here, into an extension that conforms to the protocols it needs:

```swift
extension ManageOfflineRegionsExample_Swift: UITableViewDelegate, UITableViewDataSource {
    // table view methods here
}
```